Bio/SeqAnalysis/SeqVisualizer.py

```diff
@@ -0,0 +1,679 @@
+        :param out_name: name for plot (if not passed, default name will be applied)
+
+        :return: plot showed interactively or saved as .png
+        """
+
+        if not save:
+            plt.show()
+        else:
+            plt.tight_layout()
+            plt.savefig(f'{out_name}.png')
+        self.ids = self.main_dict['id']
+
+    def show_or_save_plot_interactive(self, save, figure, out_name):
+        """
+        Method used to show (by default) or save plots.
+
+        :param save: if passed, plot will be saved in .html format instead of being shown
+        :param figure: figure object to save
+        :param out_name: name for plot (if not passed, default name will be applied)
+
+        :return: plot showed interactively or saved as .html
+        """
+        if not save:
+            figure.show()
+        else:
+            figure.write_html(f'{out_name}.html')
+        self.ids = self.main_dict['id']
+
+    def bar(self, single_seq=None, parameter=None, save=False, size=(15, 8), x_ax_title_fontsize=14,
+            y_ax_title_fontsize=14, suptitle_fontsize=13, add_legend=True, leg_size=13, out=None):
+        """
+        Method creating bar plot for chosen input parameter. Can work in single sequence mode.
+            For parameters:
+            - Sequence length
+            - Molecular weight
+            - Theoretical isoelectric point
+            - Aromaticity
+            - Flexibility
+            - Average hydrophobicity and hydrophilicity (GRAVY)
+            each bar represents sequence
+
+            For parameter:
+            - Secondary structure fraction
+            each bar represents secondary structure fraction
+
+            For parameter:
+            - Extinction coefficient
+            each bar represents cystein groups
+
+            For parameters:
+            - Amino acid composition
+            - Amino acid percent composition
+            - Frequency of C terminus
+            - Frequency of N terminus
+            each bar represents amino acids
+
+        :param single_seq: str; ID of single sequence from input dictionary; if passed, only this sequenced will be plotted
+        :param parameter: str; parameter, which values will be represented
+        :param save: boolean; if passed, plot will be saved (if not passed, plot is shown by default), by default False
+        :param size: tuple; size of plot, passed as tuple eg. (12, 8)
+        :param x_ax_title_fontsize: int; size of font of x axis title
+        :param y_ax_title_fontsize: int; size of font of y axis title
+        :param suptitle_fontsize: int; size of font of plot title
+        :param add_legend: boolean; show legend, by default True
+        :param leg_size: int; size of legend
+        :param out: str; name of output file
+
+        :return: plot shown or saved
+        """
+        if parameter is None or parameter not in self.all_args:
+            print('Pass a parameter to plot properly.')
```

Comment:
You print message about wrong parameter but do nothing? It'll just crash, do something, set parameter to default value / return error / raise exception (if not handled, should be described in docstring)